crates/bevy_ecs/src/system/commands/entity_command.rs

```diff
@@ -254,8 +254,8 @@ pub fn retain<T: Bundle>() -> impl EntityCommand {
 ///
 /// # Note
 ///
-/// This won't clean up external references to the entity (such as parent-child relationships
-/// if you're using `bevy_hierarchy`), which may leave the world in an invalid state.
+/// This will also despawn the entities in any [`RelationshipSources`](crate::relationship::RelationshipSources) that are configured
+/// to despawn descendants. For example, this will recursively despawn [`Children`](crate::hierarchy::Children).
 pub fn despawn() -> impl EntityCommand {
```

Comment:
Would it be worth adding `despawn_recursive` here so it can be marked as deprecated? Or maybe copy over the original trait and mark that?

Reply:
Good call. I added the methods directly to EntityCommands / EntityWorldMut. The trait path is already broken, so I don't see much sense in keeping it around. This feels like a reasonable middle ground.